src/RadixDlt.NetworkGateway.DataAggregator/Services/LedgerConfirmationService.cs

```diff
@@ -457,6 +460,7 @@ private void UpdateRecordsOfTopOfLedger(TransactionSummary topOfLedger)
         _observers.ForEach(x => x.RecordTopOfDbLedger(topOfLedger.StateVersion, topOfLedger.RoundTimestamp));
 
         _systemStatusService.SetTopOfDbLedgerNormalizedRoundTimestamp(topOfLedger.NormalizedRoundTimestamp);
+        _transactionService.SetCurrentEpoch(topOfLedger.Epoch);
```

Comment:
Please don't... this code is becoming more and more entangled. Ledger confirmation service instructing completely unrelated pending TX service about current epoch? This one should be decoupled.

Reply:
Yeah this was a bit of a 4am "make it work". Would you be up for me registering the PendingTransactionService as a `LedgerStateTipObserver`?